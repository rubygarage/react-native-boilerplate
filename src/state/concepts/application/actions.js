import * as types from './types';

export const setOnboardingCompletion = (isCompleted) => ({
  type: types.SET_ONBOARDING_COMPLETION,
  isCompleted,
});

export const startListenConnectionState = () => ({
  type: types.START_LISTEN_CONNECTION_STATE,
});

export const stopListenConnectionState = () => ({
  type: types.STOP_LISTEN_CONNECTION_STATE,
});

export const setConnectionState = (isConnected) => ({
  type: types.SET_CONNECTION_STATE,
  isConnected,
});
