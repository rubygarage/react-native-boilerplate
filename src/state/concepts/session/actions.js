import * as types from './types';

export const setTokens = (tokens) => ({
  type: types.SET_TOKENS,
  tokens,
});

export const removeTokens = () => ({
  type: types.SET_TOKENS,
  tokens: null,
});

export const setUserId = (id) => ({
  type: types.SET_USER_ID,
  id,
});
