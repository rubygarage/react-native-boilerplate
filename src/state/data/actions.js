import * as types from './types';

export const dataApiRequest = ({ endpoint }) => ({
  type: types.DATA_API_REQUEST,
  endpoint,
});

export const dataApiSuccess = ({ endpoint, response }) => ({
  type: types.DATA_API_SUCCESS,
  endpoint,
  response,
});

export const dataApiFailure = ({ endpoint }) => ({
  type: types.DATA_API_FAILURE,
  endpoint,
});

export const dataDelete = ({ kind, ids }) => ({
  type: types.DATA_DELETE,
  kind,
  ids,
});
