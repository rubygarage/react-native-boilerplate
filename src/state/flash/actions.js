import { ALERT_TYPES } from 'constants/alert';
import * as types from './types';

export const showFlash = ({
  id,
  lifetime,
  priority,
  isSingleShot = false,
  flashProps,
}) => ({
  type: types.SHOW_FLASH,
  flash: {
    id,
    lifetime,
    priority,
    isSingleShot,
    flashProps,
  },
});

export const showErrorFlash = (action) => ({
  type: types.SHOW_FLASH,
  flash: {
    isSingleShot: false,
    flashProps: {
      title: { id: 'shared.oops' },
      type: ALERT_TYPES.error,
      action,
      actionTextId: 'shared.tryAgain',
      isCloseable: true,
    },
  },
});

export const addFlash = (flash) => ({
  type: types.ADD_FLASH,
  flash,
});

export const hideFlash = (id) => ({
  type: types.HIDE_FLASH,
  id,
});

export const removeFlash = (id, isSingleShot) => ({
  type: types.REMOVE_FLASH,
  id,
  isSingleShot,
});
