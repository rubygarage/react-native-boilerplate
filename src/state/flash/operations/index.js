import hideFlash from './hideFlash';
import showFlash from './showFlash';

export default [
  hideFlash, showFlash,
];
