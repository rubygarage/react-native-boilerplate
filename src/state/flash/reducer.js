import { omit, append } from 'ramda';
import { combineReducers } from 'redux';

import * as types from './types';

const list = (state = {}, action) => {
  switch (action.type) {
    case types.ADD_FLASH: {
      const { flash } = action;
      const id = flash.id || `${Date.now()}_${Math.floor(Math.random() * 9999)}`;
      const priority = flash.priority || Date.now();
      return {
        ...state,
        [id]: { ...flash, id, priority },
      };
    }
    case types.REMOVE_FLASH:
      return omit([action.id], state);
    default:
      return state;
  }
};

const viewedIds = (state = [], action) => {
  switch (action.type) {
    case types.REMOVE_FLASH:
      return action.isSingleShot ? append(action.id, state) : state;
    default:
      return state;
  }
};

export default combineReducers({
  list, viewedIds,
});
