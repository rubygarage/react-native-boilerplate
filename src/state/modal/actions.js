import * as types from './types';

export const showModal = ({ modal, modalProps }) => ({
  type: types.SHOW_MODAL,
  modal,
  modalProps,
});

export const hideModal = () => ({
  type: types.HIDE_MODAL,
});
