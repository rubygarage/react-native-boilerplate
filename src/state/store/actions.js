import * as types from './types';

// eslint-disable-next-line import/prefer-default-export
export const resetStore = () => ({
  type: types.RESET_STORE,
});
