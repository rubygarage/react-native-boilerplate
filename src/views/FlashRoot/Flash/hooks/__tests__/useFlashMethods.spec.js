import { renderHook, act } from '@testing-library/react-hooks';

import HooksTestHelper from 'utils/testHelpers/hooksTestHelper';
import { hideFlash } from 'state/flash/actions';

import useFlashMethods from '../useFlashMethods';

jest.useFakeTimers();

describe('doAnimationHook', () => {
  const dispatch = jest.fn();

  HooksTestHelper.mockUseDispatchImplementation(() => dispatch);

  const props = {
    id: 'some',
    flashProps: { action: { type: 'TEST' } },
    doAnimation: jest.fn(),
    lifetime: null,
  };

  it('matches snapshot', () => {
    const { result } = renderHook(() => useFlashMethods(props));

    expect(result.current).toMatchSnapshot();
  });

  it('checks onHideFlash method', () => {
    const { result } = renderHook(() => useFlashMethods(props));

    act(() => {
      result.current.onHideFlash();
    });

    expect(dispatch).toHaveBeenCalledWith(hideFlash(props.id));
    expect(props.doAnimation).toHaveBeenCalled();
  });

  it('checks onActionClick method', () => {
    dispatch.mockClear();

    const { result } = renderHook(() => useFlashMethods(props));

    act(() => {
      result.current.onActionClick();
    });

    expect(dispatch).toHaveBeenCalledTimes(2);
    expect(dispatch).toHaveBeenNthCalledWith(1, hideFlash(props.id));
    expect(dispatch).toHaveBeenNthCalledWith(2, { type: 'TEST' });
  });

  describe('checks didMount method', () => {
    it('should call setInterval', () => {
      const { result } = renderHook(() => useFlashMethods({ ...props, lifetime: 1000 }));

      let unMountingCallback = null;

      act(() => {
        unMountingCallback = result.current.didMount();
      });

      expect(setInterval).toHaveBeenCalledWith(result.current.onHideFlash, 1000);

      unMountingCallback();

      expect(clearInterval).toHaveBeenCalled();
    });

    it('should not call setInterval', () => {
      const { result } = renderHook(() => useFlashMethods(props));

      act(() => {
        result.current.didMount();
      });

      expect(setInterval).not.toHaveBeenCalledWith();
    });
  });
});
