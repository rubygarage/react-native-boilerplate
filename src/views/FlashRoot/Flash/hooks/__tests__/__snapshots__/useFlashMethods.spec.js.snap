// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doAnimationHook matches snapshot 1`] = `
Object {
  "didMount": [Function],
  "onActionClick": [Function],
  "onHideFlash": [Function],
}
`;
